src/QMCHamiltonians/MomentumEstimator.cpp

```diff
@@ -229,36 +218,37 @@ bool MomentumEstimator::putSpecial(xmlNodePtr cur, ParticleSet& elns, bool rootN
     {
       for (int k=-kgrid; k<(kgrid+1); k++)
       {
-        PosType kpt;
-        kpt[0]=i-twist[0];
-        kpt[1]=j-twist[1];
-        kpt[2]=k-twist[2];
+        PosType ikpt,kpt;
+        ikpt[0]=i-twist[0];
+        ikpt[1]=j-twist[1];
+        ikpt[2]=k-twist[2];
+        kpt=ikpt;
```

Comment:
No need this line. Overwritten 2 lines later.

Reply:
ikpt is used in the directional. So, instead of (i-twist[0])*(i-twist[0]) in the if I use ikpt[0]*ikpt[0]. kpt is the ikpt in Lattice.k_cart with 2*pi included, which is used in the push_pack and in the sphere case.

Reply:
Yes.